### PR TITLE
Use TCP_NODELAY for improved latency

### PR DESCRIPTION
**Pull request by protoss-player**
According to the NSQ protocol, consumers have to send `FIN`s for every
message they receive, which makes them extremely slow with low `max_in_flight`
values due to TCP buffering.

Adding TCP_NODELAY option on `async.AsyncConn`'s sockets makes `nsq.Reader`
substantially faster. My tests show increase from ~50 to ~4.5K messages per
second on localhost with `max_in_flight = 1`.

----

NODELAY will be enabled by default. It may be disabled by passing `nodelay=False` to `Reader`/`Writer` ctor's kwargs.

### CODE REVIEW

**Comment by ploxiln** [comment]
Do you think there is ever real downside to enabling nodelay for nsq tcp connections? (Is it ever common for pynsq to do a sequence of small writes that go out as separate packets without nodelay?) Perhaps, instead of adding a performance tuning-option, it should just be enabled?

**Comment by mreiferson** [comment]
@ploxiln I was thinking the same (and was actually surprised this wasn't already the case)

👍 to on by default

**Comment by protoss-player** [comment]
@ploxiln I don't think NODELAY could break anything, especially when communicating over LAN. BTW, it's enabled by default for all TCP sockets in Asyncio, see https://github.com/python/cpython/blob/master/Lib/asyncio/selector_events.py#L689.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,7 @@ dist
 pynsq.egg-info
 build
 docs/_build
+.cache
+.eggs
+*.dat
+*.tmp
```
Comment (ploxiln) [inline .gitignore]: this change is not really related to the rest

**Comment by ploxiln** [comment]
To clarify, in addition to "on by default", I was proposing not adding any option to disable it.

File: .gitignore (hunk 1 above)
Comment (protoss-player) [inline .gitignore]: Yep, indeed, but running tests produced too much crap. Not sure if this could make a decent PR on its own.

File: nsq/async.py [hunk 2]
```diff
@@ -128,6 +128,8 @@ class AsyncConn(event.EventedMixin):
     :param msg_timeout: the amount of time (in seconds) that nsqd will wait
         before considering messages that have been delivered to this
         consumer timed out (requires nsqd 0.2.28+)
+
+    :param nodelay: set TCP NODELAY mode on socket
```
Comment (mreiferson) [inline nsq/async.py]: as per @ploxiln's suggestion, let's drop the option altogether!

Comment (protoss-player) [inline nsq/async.py]: OK, done

**Comment by mreiferson** [comment]
@protoss-player would you mind squashing down to one commit please?

**Comment by protoss-player** [comment]
@mreiferson Done.

BTW, I believe there's a "squash merge" option for PRs (some changes to repository settings may be required), see https://github.com/blog/2141-squash-your-commits. Never used it myself though.

**Comment by mreiferson** [comment]
I'm going to merge this, but it looks like our travis bootstrapping is broken and all tests are failing.

I'll fix separately.

Thanks for the contribution!
